src/bidiMapper/modules/context/BrowsingContextImpl.ts

```diff
@@ -695,7 +698,10 @@ export class BrowsingContextImpl {
     });
 
     this.#cdpTarget.cdpClient.on('Page.javascriptDialogOpening', (params) => {
-      if (this.id !== params.frameId) {
+      // Checking for `params.frameId` for comptaibility with Chrome
+      // versions that do not have a frameId. TODO: remove once
+      // https://crrev.com/c/6487891 is in stable.
+      if (params.frameId && this.id !== params.frameId) {
```

Comment:
This will cause duplicated events for pages with same-process iFrames. We need to check if the parent's CDP client is different from the current one to avoid duplicates.

Reply:
Something like this?